src/azul/terraform.py

```diff
@@ -676,6 +676,32 @@ def tf_config(self, app_name):
                 key.replace('.', '_'): value
                 for key, value in bucket_notifications.items()
             }
+            # To prevent a race condition by Terraform, we make the bucket
+            # notifications depend on the related aws_lambda_permission and
+            # aws_lambda_function resources.
+            # https://stackoverflow.com/questions/67010382/terraform-error-putting-s3-notification-configuration-invalidargument-unable
+            for notification in resources['aws_s3_bucket_notification'].values():
+                lambda_arns = [
+                    lambda_function['lambda_function_arn']
+                    for lambda_function in notification['lambda_function']
+                ]
+                lambda_resources = []
+                permission_resources = []
+                for key, value in resources['aws_lambda_permission'].items():
+                    lambda_arn = value['function_name']
+                    if lambda_arn in lambda_arns:
+                        assert lambda_arn.startswith('${'), lambda_arn
+                        assert lambda_arn.endswith('.arn}'), lambda_arn
+                        lambda_resources.append(lambda_arn[2:-5])
```

Comment:
Personally I prefer

```suggestion
                        match = re.fullmatch(r'\$\{(.*)\.arn\}')
                        assert match is not None
                        lambda_resources.append(match.group(1))
```

to avoid the hardcoded slice indices. But regex is definitely an overpowered tool to be using here, so feel free to disregard or come up with something else.